BlazorLeaflet/BlazorLeaflet/Models/Layer.cs

```diff
@@ -11,6 +11,18 @@ public abstract class Layer
         /// </summary>
         public string Id { get; }
 
+        /// <summary>
+        /// We need JSRuntime to call methods on the corresponding javascript object
+        /// It will be set my LeafletMap when the layer is added to the actual map.
+        /// </summary>
+        public IJSRuntime JSRuntime { get; internal set; }
+
+        /// <summary>
+        /// We need MapId to call methods on the corresponding javascript object.
+        /// It will be set my LeafletMap when the layer is added to the actual map.
+        /// </summary>
+        public string MapId { get; internal set; }
+
```

Comment:
It would be better to store the reference to a Map object inside the Layer rather than sparsely storing some bits of map data inside it.

Reply:
I agree but when a layer with an attached map object is passed to `LeafletInterops.AddLayer(...)` method the jsInteropt tries to serialize the layer as json and fails with an exception: 

    System.Text.Json.JsonException: 
    A possible object cycle was detected which is not supported. 
    This can either be due to a cycle or if the object depth is larger than the maximum allowed depth of 32

I am not very good at debugging Blazor WebAssembly. So unfortunately the solution with the `MapId` is the only one i could make work.

Please let me know if you have an idea on how to handle this problem in a better way.

Thank you for your effort!

Reply:
This normally happens because the Layer has a reference to the Map and the Map has a reference to a list of Layers which reference the same Map, so when the serializer stars doing its job there is a cyclic reference that keeps on going.

So far there is not an elegant solution because System.Text.Json doesn't let us configure how it should handle cyclic references, like Newtonsoft.Json. There is an issue on going which should solve this later https://github.com/dotnet/corefx/issues/41002.

For now I think your solution will do and this will be changed in a future refactoring when we'll have more control over the way serialization works.

Thanks for this awesome PR!

Reply:
Thank you, too! More will possibly follow when i use the Library more.